core/src/main/java/org/wso2/carbon/testgrid/core/TestPlanExecutor.java

```diff
@@ -123,7 +123,7 @@ private TestPlan destroyInfrastructure(Infrastructure infrastructure, TestPlan t
      * @return the status of the execution (success / fail)
      * @throws TestPlanExecutorException If something goes wrong while executing the TestPlan.
      */
-    public TestPlan runTestPlan(TestPlan testPlan, Infrastructure infrastructure) throws TestPlanExecutorException {
+    public TestPlan runTestPlan(TestPlan testPlan, Infrastructure infrastructure) throws TestPlanExecutorException, UnsupportedProviderException {
```

Comment:
Catch the UnsupportedProviderException inside this and throw a TestPlanExecutorException after wrapping it.